tests/test_model.py

```diff
@@ -1,297 +1,235 @@
     model = MrVI(adata)
-    model.train(2, check_val_every_n_epoch=1, train_size=0.5)
-    donor_keys = ["meta1_cat", "meta2", "cont_cov"]
-    model.perform_multivariate_analysis(
-        donor_keys=donor_keys, store_lfc=True, add_batch_specific_offsets=False
-    )
+    model.train(1, check_val_every_n_epoch=1, train_size=0.5)
     model.get_local_sample_distances()
-
-    MrVI.setup_anndata(
-        adata,
-        sample_key="sample",
-        batch_key="batch",
-        continuous_covariate_keys=["cont_cov"],
-    )
+    model.get_local_sample_distances(normalize_distances=True)
+    model.get_latent_representation(give_z=False)
+    model.get_latent_representation(give_z=True)
+
+
+@pytest.mark.parametrize(
+    "setup_kwargs, de_kwargs",
+    [
+        (
+            {"sample_key": "sample", "batch_key": "batch"},
+            [
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "add_batch_specific_offsets": True,
+                },
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "lambd": 1e-1,
+                    "add_batch_specific_offsets": True,
+                },
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "filter_inadmissible_samples": True,
+                    "add_batch_specific_offsets": True,
+                },
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "add_batch_specific_offsets": False,
+                },
+            ],
+        ),
+        (
+            {
+                "sample_key": "sample",
+                "batch_key": "batch",
+                "continuous_covariate_keys": ["cont_cov"],
+            },
+            [
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "add_batch_specific_offsets": False,
+                }
+            ],
+        ),
+        (
+            {"sample_key": "sample", "batch_key": "dummy_batch"},
+            [
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                },
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "lambd": 1e-1,
+                },
+                {
+                    "sample_cov_keys": ["meta1_cat", "meta2", "cont_cov"],
+                    "store_lfc": True,
+                    "filter_inadmissible_samples": True,
+                },
+            ],
+        ),
+    ],
+)
```

Comment:
This is lowkey kind of hard to parse, is there a way to simplify it?

Reply:
ok nvm I see what it's doing now

Reply:
yea..it's because its replicating the old behavior, which was not neatly combinatorial way, but I did not want to significantly change what was being tested. We can refactor it again if needed and it will be easier this time with parametrize